app/controllers/users/sessions_controller.rb

```diff
@@ -13,11 +13,17 @@
 class Users::SessionsController < Devise::SessionsController
   before_action :configure_sign_in_params, only: :create
 
-  prepend_before_action :authenticate_with_otp_two_factor,
+  prepend_before_action :handle_otp_flow,
     if: -> { action_name == "create" && otp_two_factor_enabled? }
 
   protect_from_forgery with: :exception, prepend: true, except: :destroy
 
+  def edit_mobile_number
+    @user = self.resource = find_user
```

Comment:
why do we need to set `self.resource` here?

Reply:
Because the views are still in `devise/controllers` and use the Devise `resource` form rather than an explicit `@user`. That said, there's nothing to say we can't just move those to `users/session` now and use `@user`.

Reply:
Ah, interestingly, when I removed this and ran `rspec spec/features/staff/users_can_sign_in_spec.rb` then all the tests still passed. Is this because the tests don't cover something?

Reply:
It's because in Devise they're [synonymous](https://github.com/heartcombo/devise/blob/main/app/controllers/devise_controller.rb#L38-L40)

Reply:
(I removed the `@user = ` part this morning)